coi/src/embedding.rs

```diff
@@ -94,82 +69,25 @@ where
 ///
 /// See [`pairwise_cosine_similarity`] for details.
 pub fn cosine_similarity(a: ArrayView1<'_, f32>, b: ArrayView1<'_, f32>) -> f32 {
-    let norm_a = l2_norm(a.view());
-    if norm_a <= 0. {
-        return 1.;
-    }
-
-    let norm_b = l2_norm(b.view());
-    if norm_b <= 0. {
-        return 1.;
-    }
-
-    (a.dot(&b) / norm_a / norm_b).clamp(-1., 1.)
+    a.dot(&b).clamp(-1., 1.)
```

Comment:
:art:/:wrench: with the proposed changes to `normalized()`, we would need to map the edge case here, such that we still have the desired property such that the zero vector is similar to all other vectors (in reality this should never occur anyways), we have to be careful though that we don't confuse orthogonal vectors with the all-zero vector
```suggestion
    if a.iter().any(|&v| v != 0.) || b.iter().any(|&v| v != 0.) {
        a.dot(&b).clamp(-1., 1.)
    } else {
        1.
    }
```
since the check is short-circuiting, it should be very cheap in almost all cases.

Reply:
yup, wasn't really sure in these cases, will take your snippet